numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,18 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    def __new__(cls, pyfunc=None, *args, **kw):
+        def new_with_kw(pyfunc):
+            ''' ``vectorize`` with presupplied keyword arguments. '''
+            self = object.__new__(cls)
+            self.__init__(pyfunc, *args, **kw)
```

Comment:
Oops, playing around with `__new__` some more I think this actually ends up calling `__init__` twice in the case where `pyfunc` is supplied. (apparently, `__init__` gets called conditionally on the output of `__new__` based on its type)  It doesn't break any tests but I should fix this